x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/components/description_step/index.tsx

```diff
@@ -111,14 +110,14 @@ export const addFilterStateIfNotThere = (filters: Filter[]): Filter[] => {
 const getDescriptionItem = (
   field: string,
```

Comment:
After reading through the comments below, I would suggest for maintainers and future additions to the rules editor and maintenance what we do (or put a TODO block for the next person) is that this field isn't a string but rather an exact string like so:

```ts
type FieldValues = 'threat' | 'queryBar' | 'references' | ...;
```

And then below this would become an exhaustive switch like so:

```ts
  switch (field) {
    case 'queryBar': {
    }
  }
```

Up to you on these parts but I would at least add a TODO block or a pointer for the next maintainer. This would become less error prone and more safe code.

Reply:
Deferred to https://github.com/elastic/kibana/issues/60567